system/src/main/java/com/percussion/servlets/PSLoginServlet.java

```diff
@@ -353,18 +374,27 @@ protected static boolean isValidRedirectUri(HttpServletRequest request, String u
     * @throws ServletException
     */
    private void authenticate(HttpServletRequest request,
-                             HttpServletResponse response, String uid, String pwd) throws IOException,
+                             HttpServletResponse response, String uid, String pwd,boolean legacyUI) throws IOException,
            ServletException
    {
       try
       {
 
          HttpSession sess = request.getSession(true);
-//         event.setSessionid(sess.getId());
 
          String redirect = (String) sess.getAttribute(REDIRECT_URL);
-         if (redirect == null)
-            redirect = RHYTHMYX_INDEX_PAGE;
+         if (redirect == null) {
+            if(!legacyUI) {
+               redirect = CMS_INDEX_PAGE;
+            }else{
+               redirect = LEGACY_INDEX_PAGE;
+            }
+
+         }else{
+            if(legacyUI) {
+               redirect = LEGACY_INDEX_PAGE;
+            }
+         }
```

Comment:
we are not setting variable "redirect" value  incase not legacyUI (else condition is missing)

Reply:
That is covered on line 385.  We only override the redirectUrl if the legacy UI checkbox is checked.

Reply:
that is String redirect = (String) sess.getAttribute(REDIRECT_URL);

so, can sess.getAttribute(...) may return null... just in case? that's why I had comment.

Reply:
Yes it can return null - that is why the null check on 386      if (redirect == null) {

Reply:
sorry, missed... correct.